plugins/dbviewer/api/api.js

```diff
@@ -353,7 +365,22 @@ var common = require('../../../api/utils/common.js'),
 
         var validateUserForWriteAPI = ob.validateUserForWriteAPI;
         validateUserForWriteAPI(function() {
-            if ((params.qstring.dbs || params.qstring.db) && params.qstring.collection && params.qstring.document && params.qstring.collection.indexOf("system.indexes") === -1 && params.qstring.collection.indexOf("sessions_") === -1) {
+            if ((params.qstring.dbs || params.qstring.db) && params.qstring.collection && params.qstring.action === 'get_indexes') {
+                if (params.member.global_admin) {
+                    getIndexes();
+                }
+                else {
+                    dbUserHassAccessToCollection(params.qstring.collection, function(hasAccess) {
+                        if (hasAccess) {
+                            getIndexes();
+                        }
+                        else {
+                            common.returnMessage(params, 401, 'User does not have right to view this collection');
+                        }
+                    });
+                }
+            }
+            else if ((params.qstring.dbs || params.qstring.db) && params.qstring.collection && params.qstring.document && params.qstring.collection.indexOf("system.indexes") === -1 && params.qstring.collection.indexOf("sessions_") === -1) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is too long.](https://app.codacy.com/app/ar2rsawseen/countly-server/pullRequest?prid=4498669)